crates/sui-graphql-rpc/src/server/builder.rs

```diff
@@ -69,17 +76,20 @@ impl ServerBuilder {
         self
     }
 
+    #[allow(clippy::redundant_clone)]
     pub fn build(self) -> Server {
         let address = self.address();
         let schema = self.schema.finish();
 
         let app = axum::Router::new()
             .route("/", axum::routing::get(graphiql).post(graphql_handler))
-            .layer(axum::extract::Extension(schema))
+            .layer(axum::extract::Extension(schema.clone()))
             .layer(middleware::from_fn(check_version_middleware))
             .layer(middleware::from_fn(set_version_middleware));
         Server {
             server: axum::Server::bind(&address.parse().unwrap()).serve(app.into_make_service()),
+            #[cfg(test)]
+            schema,
         }
     }
 }
```

Comment:
So, instead of this, have this:

```rust
fn build_schema(self) -> Schema<(Query, EmptyMutation, EmptySubscription)> {
    self.schema.finish()
}

pub fn build(self) -> Server {
    let address = self.address();
    let schema = self.build_schema();

    let app = axum::Router::new()
        .route("/", axum::routing::get(graphiql).post(graphiql_handler))
        .layer(axum::extract::Extension(schema))
        .layer(middleware::from_fn(check_version_middleware))
        .layer(middleware::from_fn(set_version_middleware));

    Server {
        server: axum::server::bind(&address.parse().unwrap()).serve(app.into_make_service()),
    }
}
```

And then call `biuld_schema` instead of `build` in the relevant tests.